docs/contributing-to-airbyte/tutorials/building-a-java-destination.md

```diff
@@ -0,0 +1,175 @@
+
+Your generated template should have the spec file in `airbyte-integrations/connectors/destination-<name>/src/main/resources/spec.json`. Edit it and you should be done with this step. 
+
+For more details on what the spec is, you can read about the Airbyte Protocol [here](../../understanding-airbyte/airbyte-specification.md).
+
+{% hint style="info" %}
+The generated code implements the `spec` method for you. As long as there's a file called `spec.json` in the directory mentioned above, the generated connector will take care of reading it and converting 
+{% endhint %}
+
+See the `spec` operation in action:  
+```bash
+# First build the connector 
+./gradlew :airbyte-integrations:connectors:destination-<name>:build
+
+# Run the spec operation
+docker run --rm airbyte/destination-<name>:dev spec
+```
+
+### Step 4: Implement `check`
+
+The check operation accepts a JSON object conforming to the `spec.json`. In other words if the `spec.json` said that the destination requires a `username` and `password` the config object might be `{ "username": "airbyte", "password": "password123" }`. It returns a json object that reports, given the credentials in the config, whether we were able to connect to the destination. 
+
+While developing, we recommend storing any credentials in `secrets/config.json`. Any `secrets` directory in the Airbyte repo is gitignored by default.
+
+Implement the `check` method in the generated file `<Name>Destination.java`. Here's an [example implementation](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java#L94) from the BigQuery destination.  
+
+Verify that the method is working by placing your config in `secrets/config.json` then running: 
+
+```
+# First build the connector
+./gradlew :airbyte-integrations:connectors:destination-<name>:build
+
+# Run the check method
+docker run -v $(pwd)/secrets:/secrets --rm airbyte/destination-<name>:dev check --config /secrets/config.json
+```
+
+### Step 5: Implement `write`
+The `write` operation is the main workhorse of a destination connector: it reads input data from the source and writes it to the underlying destination. It takes as input the config file used to run the connector as well as the configured catalog: the file used to describe the schema of the incoming data and how it should be written to the destination. 
+
+To implement the `write` Airbyte operation, implement the `getConsumer` method in your generated `<Name>Destination.java` file. Here are some example implementations from different destination conectors:
+ 
+* [BigQuery](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java#L188) 
```

Comment:
Thinking out loud: is it worth leaving an info section here about the Abstract Destination concept for destinations supported by JDBC?

Reply:
BQ is not JDBC - i added PG as an example of JDBC